src/io/webcam.rs

```diff
@@ -0,0 +1,247 @@
+use crate::image::{Image, ImageSize};
+use anyhow::Result;
+use gst::prelude::*;
+
+/// A builder for creating a WebcamCapture object
+pub struct WebcamCaptureBuilder {
+    camera_id: usize,
+    size: Option<ImageSize>,
+}
+
+impl WebcamCaptureBuilder {
+    /// Creates a new WebcamCaptureBuilder object with default values.
+    ///
+    /// Note: The default camera id is 0 and the default image size is None
+    ///
+    /// # Returns
+    ///
+    /// A WebcamCaptureBuilder object
+    pub fn new() -> Self {
+        Self {
+            camera_id: 0,
+            size: None,
+        }
+    }
+
+    /// Sets the camera id for the WebcamCaptureBuilder.
+    ///
+    /// # Arguments
+    ///
+    /// * `camera_id` - The desired camera id
+    pub fn camera_id(mut self, camera_id: usize) -> Self {
+        self.camera_id = camera_id;
+        self
+    }
+
+    /// Sets the image size for the WebcamCaptureBuilder.
+    ///
+    /// # Arguments
+    ///
+    /// * `size` - The desired image size
+    pub fn with_size(mut self, size: ImageSize) -> Self {
+        self.size = Some(size);
+        self
+    }
+
+    /// Create a new [`WebcamCapture`] object.
+    pub fn build(self) -> Result<WebcamCapture> {
+        WebcamCapture::new(self.camera_id, self.size)
+    }
+}
+
+impl Default for WebcamCaptureBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+/// A webcam capture object that grabs frames from the camera
+/// using GStreamer.
+///
+/// # Example
+///
+/// ```no_run
+/// use kornia_rs::{image::ImageSize, io::webcam::WebcamCaptureBuilder};
+///
+/// fn main() -> Result<(), Box<dyn std::error::Error>> {
+///   // create a webcam capture object with camera id 0
+///   // and force the image size to 640x480
+///   let webcam = WebcamCaptureBuilder::new()
+///     .camera_id(0)
+///     .with_size(ImageSize {
+///       width: 640,
+///       height: 480,
+///   })
+///   .build()?;
+///
+///   // start grabbing frames from the camera
+///   webcam.run(|img| {
+///     println!("Image: {:?}", img.size());
+///   })?;
+///
+///   Ok(())
+/// }
+/// ```
+pub struct WebcamCapture {
+    pipeline: gst::Pipeline,
+    receiver: std::sync::mpsc::Receiver<Image<u8, 3>>,
+    handle: Option<std::thread::JoinHandle<()>>,
+}
+
+impl WebcamCapture {
+    /// Creates a new WebcamCapture object.
+    ///
+    /// # Arguments
+    ///
+    /// * `camera_id` - The camera id used for capturing images
+    /// * `size` - The image size used for resizing directly from the camera
+    ///
+    /// # Returns
+    ///
+    /// A WebcamCapture object
+    fn new(camera_id: usize, size: Option<ImageSize>) -> Result<Self> {
```

Comment:
to an extent a matter of taste, but I'd only reach to `anyhow::Result` definitely no longer to be handled programmatically and 100% a user or log facing message. The problem with anyhow is that one can't match on the different error outcomes. Instead, I prefer `thiserror` errors whenever it makes sense, and I think here it would be a good fit.

Reply:
i've been [experimenting](https://github.com/kornia/kornia-rs/compare/this-error?expand=1) a bit with thiserror and i feel  that is a bit impractical here because i have to be mapping all the time all the defined errors by the methods of the other libraries. It give more semantics yes, but seems a bit of a blocker for development at least now. And specially because this crate is more an aggregation of libraries, and thiserror seems pretty well defined when you start a very foundational crate with almost none dependencies.